documentation/coding_guidelines.md

```diff
@@ -286,38 +290,70 @@ in tests/benchmarks](https://github.com/rust-lang/cargo/issues/2911), we rely on
 conditions to perform this conditional compilation:
 - the test flag, which is activated by dependent test code in the same crate
   as the conditional test-only code.
-- the "fuzzing" custom feature, which is used to enable fuzzing and testing
+- the `fuzzing` custom feature, which is used to enable fuzzing and testing
 related code in downstream crates. Note that this must be passed explicitly to
 `cargo xtest` and `cargo x bench`. Never use this in `[dependencies]` or
 `[dev-dependencies]` unless the crate is only for testing, otherwise Cargo's
 feature unification may pollute production code with the extra testing/fuzzing code.
 
-As a consequence, it is recommended that you set up your test-only code in the following fashion. For the sake of example, we'll consider you are defining a test-only helper function `foo` in `foo_crate`:
-1. Define the "testing" flag in `foo_crate/Cargo.toml` and make it non-default:
-    ```
+As a consequence, it is recommended that you set up your test-only code in the following fashion.
+
+**For production crates:**
+
+Production crates are defined as the set of crates that create externally published artifacts, e.g. the Libra validator,
+the Move compiler, and so on.
+
+For the sake of example, we'll consider you are defining a test-only helper function `foo` in `foo_crate`:
+
+1. Define the `fuzzing` flag in `foo_crate/Cargo.toml` and make it non-default:
+    ```toml
     [features]
     default = []
     fuzzing = []
     ```
 2. Annotate your test-only helper `foo` with both the `test` flag (for in-crate callers) and the `"fuzzing"` custom feature (for out-of-crate callers):
-    ```
+    ```rust
     #[cfg(any(test, feature = "fuzzing"))]
     fn foo() { ... }
     ```
 3. (optional) Use `cfg_attr` to make test-only trait derivations conditional:
-    ```
+    ```rust
     #[cfg_attr(any(test, feature = "testing"), derive(FooTrait))]
```

Comment:
should we update "testing" to "fuzzing" here?

Reply:
Done!